### ISSUE TITLE
Javascript compilation broken

### ISSUE THREAD

**Issue by scottkunkel** [godotengine/godot#29443]
**Godot version:**

85a338295

**OS/device including version:**

win 10

**Issue description:**

scons platform=javascript target=release_debug tools=no optimize=size use_lto=yes -j8
```
JS optimizer error:
Unexpected token: operator (>) (line: 1637, col: 160, pos: 61425)

================================

 function() { return Module.resizeCanvasOnStart },
 function($0) { stringToUTF8(Module.locale, $0, 16); },
 function($0, $1, $2, $3) { const send_notification = cwrap('send_notification', null, ['number']); const notifications = arguments; (['mouseover', 'mouseleave', 'focus', 'blur']).forEach(function(event, index) { Module.canvas.addEventListener(event, send_notification.bind(null, notifications[index])); }); const update_clipboard = cwrap('update_clipboard', null, ['string']); window.addEventListener('paste', function(evt) { update_clipboard(evt.clipboardData.getData('text')); }, true); },
 function($0) { window.alert(UTF8ToString($0)); },
 function($0) { document.title = UTF8ToString($0); },
 function($0) { var text = UTF8ToString($0); if (!navigator.clipboard || !navigator.clipboard.writeText) return 1; navigator.clipboard.writeText(text).catch(e => { console.error("Setting OS clipboard is only possible from an input callback for the HTML5 plafrom. Exception:", e); }); return 0; },
                                                                                                                                                                ^
================================
```
**Steps to reproduce:**
build with emscripten in win10

**Minimal reproduction project:**

**Comment by akien-mga** [comment]
Looks like a regression from #29298 @Faless.

**Comment by scottkunkel** [comment]
should probably be (e) => {

**Comment by scottkunkel** [comment]
Looks like arrow functions are not supported here:
function(e){ instead of e => {
works

**Comment by Calinou** [comment]
According to https://github.com/emscripten-core/emscripten/pull/7973, ES2015+ JavaScript should be supported in the Emscripten optimizer pass.

**Comment by scottkunkel** [comment]
1.38.25...
sigh will update and recompile

**Comment by scottkunkel** [comment]
works with emscripten 1.38.31

**Comment by akien-mga** [comment]
Fixed by godotengine/godot-docs#2500.
